src/ModelContextProtocol.AspNetCore/StatefulSessionManager.cs

```diff
@@ -0,0 +1,242 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace ModelContextProtocol.AspNetCore;
+
+internal sealed partial class StatefulSessionManager(
+    IOptions<HttpServerTransportOptions> httpServerTransportOptions,
+    ILogger<StatefulSessionManager> logger) : ConcurrentDictionary<string, StreamableHttpSession>(StringComparer.Ordinal)
```

Comment:
Did you derive like this in order to save an allocation? Presumably there's only ever one of these? Feels a bit brittle to be exposing every method from the underlying dictionary directly.